packages/compiler-cli/src/ngtsc/typecheck/src/type_check_block.ts

```diff
@@ -1049,6 +1049,32 @@ class TcbUnclaimedOutputsOp extends TcbOp {
         continue;
       }
 
+      // Check for split two way binding with unclaimed outputs
+      if (this.tcb.env.config.checkTypeOfOutputEvents && output.name.endsWith('Change')) {
+        const inputName = output.name.slice(0, -6);
+        if (this.claimedInputs.has(inputName)) {
+          // Get input bindings from template AST. If not found, then the user
+          // did not bind an input, so this is not a two-way data binding.
+          const input = this.element.inputs.find(input => input.name == inputName);
+          if (input && input.sourceSpan == output.sourceSpan) {
```

Comment:
In the compiler we prefer to check values explicitly rather than relying on truthiness. Additionally, in JavaScript you should always prefer `===` to `==` as the latter has some really unintuitive behavior in specific cases. For example, `3 == '3'` is true.


```suggestion
          if (input !== undefined && input.sourceSpan === output.sourceSpan) {
```